### PR TITLE
REP-114 added documentation module

### LINKED ISSUE #114: Configure docs to be pulled into upstream projects
**Describe the task**
Update the documentation so that it can be pulled into upstream projects via the maven artifact.

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

adds documentation module

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?

@jaymcnallie 
@shaundmorris 
@sgalla
@rececoffin

#### How should this be tested? (List steps with links to updated documentation)

build docs with `-Prelease` flag to verify pdfs generate correctly

#### What are the relevant tickets?
Fixes #114 

#### Checklist:
- [x] Documentation Updated!
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: docs/pom.xml [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+                        </configuration>
+                        <id>include-docs</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <!-- This execution is for publishing the html file to nexus. -->
```
Comment (shaundmorris) [inline docs/pom.xml:308]: ❔ This publishes to the internal nexus, correct?

File: docs/src/main/resources/content/config.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
+Version ${project.version}. Connexta
```
Comment (shaundmorris) [inline docs/src/main/resources/content/config.adoc:1]: ❔ Do we need the Connexta here? Should we switch to Octo? Not sure how we are handling that with portions that have existed for years vs new docs/code.

File: docs/src/main/resources/content/replication-configuring.adoc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+[TIP]
+====
+Details about these commands can also be viewed from the ${command-console} using the `--help` option.
+====
+
+=== Adding a Replication Configuration
+
+The preferred method of configuring replication is through the replication UI found at https://<host>:<port>/admin/replication
+
+Configuration can still be created and managed via the commandline as detailed below.
```
Comment (shaundmorris) [inline docs/src/main/resources/content/replication-configuring.adoc:22]: @ricklarsen, we might need to check with @clockard I think some of these commands were taken out at a specific verison.

File: docs/src/main/resources/content/replication-installing.adoc [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+. Navigate to the *${admin-console}*.
+. Select the *System* tab.
+. Select the *Features* tab.
+. Locate the *replication* feature.
+. Select the *Install* arrow for the *replication* feature.
+. The status will change from *Uninstalled* to *Installed*.
+
+=== Installation When Replication _*Not*_ Included In Distribution
+[NOTE]
+For this scenario first copy the `replication.policy` file into the `<${branding}-home>/security/` directory and restart the system.
```
Comment (shaundmorris) [inline docs/src/main/resources/content/replication-installing.adoc:26]: ❔ Can we document or link to an example of this policy?

**Review by shaundmorris — APPROVED** [review]

**Comment by clockard** [comment]
@ricklarsen Just wanted you to know that the master branch of this repo isn't used by any projects I am aware of. The 0.2.x and 0.3.x branches are though. The master branch started a transition of the service but was moved to a different repo part way through. 

The documentation that was created for this repo is mostly only valid for the 0.2.x version.

File: docs/src/main/resources/content/config.adoc (hunk 2 above)
Comment (ricklarsen) [reply docs/src/main/resources/content/config.adoc:1]: > @ricklarsen Just wanted you to know that the master branch of this repo isn't used by any projects I am aware of. The 0.2.x and 0.3.x branches are though. The master branch started a transition of the service but was moved to a different repo part way through.
> 
> The documentation that was created for this repo is mostly only valid for the 0.2.x version.

thanks! I'll update for that

File: docs/src/main/resources/content/replication-configuring.adoc (hunk 3 above)
Comment (ricklarsen) [reply docs/src/main/resources/content/replication-configuring.adoc:22]: yeah. It looks like I targeted the wrong branch

File: docs/src/main/resources/content/config.adoc (hunk 2 above)
Comment (ricklarsen) [reply docs/src/main/resources/content/config.adoc:1]: > ❔ Do we need the Connexta here? Should we switch to Octo? Not sure how we are handling that with portions that have existed for years vs new docs/code.

That's a good question. I imagine we'd update it, but I'll be happy to revert if I'm wrong

**Comment by ricklarsen** [comment]
Closing to move this content to the correct branch

**Pull request closed without merging**